cmd/edv-rest/startcmd/start.go

```diff
@@ -119,16 +125,51 @@ func (s *HTTPServer) ListenAndServe(host, certFile, keyFile string, router http.
 	return http.ListenAndServe(host, router)
 }
 
+type dbInitializer interface {
+	CreateConfigStore(provider edvprovider.EDVProvider) error
+}
+
+// ActualDBInitializer enables mocking to avoid making connection to couchdb in unit testing
+type ActualDBInitializer struct{}
+
+// CreateConfigStore creates the config store for all database types, and creates indices if supported.
+func (i *ActualDBInitializer) CreateConfigStore(provider edvprovider.EDVProvider) error {
+	err := provider.CreateStore(dataVaultConfigurationStoreName)
+	if err != nil {
+		if errors.Is(err, storage.ErrDuplicateStore) {
+			return nil
```

Comment:
nit: note that the fact that the store exists doesn't mean the "reference id index" exists

Reply:
i'll create an issue and work on this george since I'm not sure how to fix this on the top of my head :)